### PR TITLE
fix lints found by golangci-lint

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary

- fix lints found by golangci-lint
the main branch have the check failing https://github.com/sigstore/sigstore/runs/5582086106?check_suite_focus=true

#### Ticket Link
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: pkg/signature/kms/aws/client.go [hunk 1]
```diff
@@ -276,14 +285,17 @@ func (a *awsClient) verifyRemotely(ctx context.Context, sig []byte, digest []byt
...
 	key, err := a.keyCache.GetByLoader(cacheKey, a.keyCacheLoaderFunctionWithContext(ctx))
 	if err != nil {
 		return nil, err
 	}
 	return key.(*cmk).PublicKey, nil
 }
-func (a *awsClient) sign(ctx context.Context, digest []byte, algorithm crypto.Hash) ([]byte, error) {
+
+// TODO: check if can remove algorithm
+func (a *awsClient) sign(ctx context.Context, digest []byte, algorithm crypto.Hash) ([]byte, error) { // nolint: unparam
```
Comment (bobcallaway) [inline pkg/signature/kms/aws/client.go:298]: I think you could change `algorithm` to `_`

File: pkg/signature/kms/azure/client.go [hunk 2]
```diff
@@ -85,7 +85,8 @@ func parseReference(resourceID string) (vaultURL, vaultName, keyName string, err
 	return
 }
 
-func newAzureKMS(ctx context.Context, keyResourceID string) (*azureVaultClient, error) {
+// TODO: remove the context?
+func newAzureKMS(ctx context.Context, keyResourceID string) (*azureVaultClient, error) { // nolint: unparam
```
Comment (bobcallaway) [inline pkg/signature/kms/azure/client.go:89]: I think you can change `ctx` to `_`

File: pkg/signature/kms/hashivault/client.go [hunk 3]
```diff
@@ -145,7 +145,8 @@ func newHashivaultClient(address, token, transitSecretEnginePath, keyResourceID
 	return hvClient, nil
 }
 
-func oidcLogin(ctx context.Context, address, path, role, token string) (string, error) {
+// TODO: remove the context?
+func oidcLogin(ctx context.Context, address, path, role, token string) (string, error) { // nolint: unparam
```
Comment (bobcallaway) [inline pkg/signature/kms/hashivault/client.go:149]: same here

**Review by bobcallaway — COMMENTED** [review]
i think the changes in your TODOs are safe to make

File: pkg/signature/kms/hashivault/client.go (hunk 3 above)
Comment (cpanato) [reply pkg/signature/kms/hashivault/client.go:149]: thanks Bob for the review and feedback

**Review by dlorenc — APPROVED** [review]
Thanks!

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**